src/Symfony/Component/DependencyInjection/Argument/ServiceLocator.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Argument;
+
+use Symfony\Component\DependencyInjection\Exception\InvalidArgumentException;
+
+/**
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+final class ServiceLocator
```

Comment:
Yes psr-11 will solve part of the problem (it solves the coupling with the symfony framework).

But let's say we use this feature for some symfony component (Form Types could be an example), if i want to use this component outside the symfony framework I would need to : require an psr-11 container, instantiate it with the services I want and then pass it to the component when I could just use a PHP array.